packages/fxa-content-server/app/scripts/models/account.js

```diff
@@ -771,6 +772,42 @@ const Account = Backbone.Model.extend(
         });
     },
 
+    /**
+     * Verify the session and account using the verification code.
+     *
+     * @param {String} code - the verification code
+     * @param {Object} [options]
+     * @param {Object} [options.service] - the service issuing signup request
+     * @returns {Promise} - resolves when complete
+     */
+    verifySessionCode(code, options = {}) {
+      const newsletters = this.get('newsletters');
+      if (newsletters && newsletters.length) {
+        this.unset('newsletters');
+        options.newsletters = newsletters;
+      }
+
+      return this._fxaClient
+        .sessionVerifyCode(this.get('sessionToken'), code, options)
+        .then(() => {
+          if (newsletters) {
+            this._notifier.trigger('flow.initialize');
+            this._notifier.trigger('flow.event', {
```

Comment:
This feels like the wrong place to emit these from, but then I checked the source and we already do the same in `verifySignUp`. So I guess it's fine?

Previously I think we tried to emit all the flow events from the view layer though, just because semantically that seems like where they should live. And if multiple views need to emit the same event, we use a mixin to eliminate duplicate code (and there's already a view mixin for `flow.initialize`). Is there some reason why that approach won't work here? (sorry I don't have all the context)